tests/test_funcs.py

```diff
@@ -112,6 +112,30 @@ def test_combination(self):
         assert_allclose(cov(y, xp=xp), xp.asarray(2.144133), rtol=1e-6)
 
 
+class TestCreateDiagonal:
+    def test_1d(self):
```

Comment:
Could we use the test from SciPy? I think it is stronger and helped to root out an issue in the original implementation?

Reply:
Yep let's do that - could also contribute it back to NumPy!